torchvision/datasets/lsun.py

```diff
@@ -23,7 +23,7 @@ def __init__(
                              readahead=False, meminit=False)
         with self.env.begin(write=False) as txn:
             self.length = txn.stat()['entries']
-        cache_file = '_cache_' + ''.join(c for c in root if c in string.ascii_letters)
+        cache_file = os.path.join(root, '_cache_' + ''.join(c for c in root if c in string.ascii_letters))
```

Comment:
It would not be better to put cache into a temp folder ?

Reply:
> It would not be better to put the cache into a temp folder ?

This cache is only ever generated once for every data file that is read and thus should per persistent.

> I also imagine a situation where the dataset is on a readonly shared storage, so creating a file there can fail as well. 

That is true, but this also applies to every other dataset where we create additional files. Not sure what our policy is about this in general.

Reply:
I agree with @vfdev-5 on this.

For datasets that we download from the internet (and thus we have write rights), it would be acceptable to add the caching be in the same location as the dataset, but if the data is already downloaded we can't guarantee that.

One idea would be to use the [caching system from torchhub](https://github.com/pytorch/pytorch/blob/e43ea227fe014c587e2141070bef4320cb07d147/torch/hub.py#L97-L102), adapted to save cache to `torchvision/datasets` or something like that.

Thoughts?

Reply:
> For datasets that we download from the internet (and thus we have write rights), it would be acceptable to add the caching be in the same location as the dataset, but if the data is already downloaded we can't guarantee that.

In general, I agree with the sentiment, but there is nothing stopping us from implementing a `download` flag for `LSUN`. In fact #3452 proposed just that. 

> One idea would be to use the caching system from torchhub, adapted to save cache to `torchvision/datasets` or something like that.

Given that `LSUN` is the only dataset that does this, I think this is a bit of an overkill. If this comes up more often in the future that might be the way to go, but until than I wouldn't do that.

Reply:
A question worth asking is what is the overhead of skipping the caching altogether. How much of a slowdown would it be?

Reply:
I'll look into it.

Reply:
I tested with `classes=["bedroom_train"]` which is by far the largest individual file. Extracting the keys took about 3 minutes. Given that we load all classes if we set `classes="train"`, we really shouldn't get rid of the caching.

Reply:
Ok, thanks for looking into this!